packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/UiThreadUtil.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.bridge
+
+import android.os.Handler
+import android.os.Looper
+import com.facebook.react.common.build.ReactBuildConfig
+ 
+/** Utility for interacting with the UI thread. */
+public object UiThreadUtil {
+ 
+    @Volatile private var sMainHandler: Handler? = null
+ 
+    private val mainHandler: Handler
+        get() {
+             if (sMainHandler == null) {
+                 synchronized(this) {
+                     if (sMainHandler == null) {
+                         sMainHandler = Handler(Looper.getMainLooper())
+                     }
+                 }
+             }
+             return sMainHandler!!
+        }
+ 
+    /** Exposed for Java interop (e.g. Java calls to `UiThreadUtil.getUiThreadHandler()`) */
+    @JvmStatic
+    public fun getUiThreadHandler(): Handler {
+         return mainHandler
+    }
+ 
+    /** @return `true` if current thread is the UI thread. */
+    @JvmStatic
+    public fun isOnUiThread(): Boolean {
+         return Looper.getMainLooper().thread == Thread.currentThread()
+    }
+ 
+    /**
+      * Throws an {@link AssertionException} if the current thread is not the UI thread. This is a
+      * noop in production, and is only meant to run in debug mode! If you need to check for
+      * incorrect-thread issues in production, duplicate this code and call it elsewhere.
+    */
+    @JvmStatic
+    public fun assertOnUiThread() {
+         if (ReactBuildConfig.DEBUG) {
+             SoftAssertions.assertCondition(isOnUiThread(), "Expected to run on UI thread!")
+         }
+    }
+ 
+    /**
+      * Throws an {@link AssertionException} if the current thread is the UI thread. This is a noop
+      * in production, and is only meant to run in debug mode! If you need to check for
+      * incorrect-thread issues in production, duplicate this code and call it elsewhere.
+      */
+    @JvmStatic
+    public fun assertNotOnUiThread() {
+         if (ReactBuildConfig.DEBUG) {
+             SoftAssertions.assertCondition(!isOnUiThread(), "Expected not to run on UI thread!")
+         }
+    }
+ 
+     /** Runs the given {@code Runnable} on the UI thread. */
```

Comment:
Could you use KDoc formatting instead of Javadoc formatting?

E.g: 

```suggestion
     /** Runs the given [Runnable] on the UI thread. */
```

Same for the others

Reply:
Done